core/domain/activity_jobs_one_off.py

```diff
@@ -235,3 +235,197 @@ def reduce(key, values):
             yield (key, values)
         else:
             yield (key, len(values))
+
+
+class AddMentionedUserIdsContentJob(jobs.BaseMapReduceOneOffJobManager):
+    """For every snapshot content of a rights model, merge the data from all
+    the user id fields in content together and put them in the
+    mentioned_user_ids field of an appropriate RightsSnapshotMetadataModel.
+    """
+
+    @staticmethod
+    def _add_collection_user_ids(snapshot_content_model):
+        """Merge the user ids from the snapshot and put them in the parent
+        collection rights model.
+        """
+        content_dict = (
+            collection_models.CollectionRightsModel.convert_to_valid_dict(
+                snapshot_content_model.content))
+        reconstituted_rights_model = (
+            collection_models.CollectionRightsModel(**content_dict))
+        snapshot_metadata_model = (
+            collection_models.CollectionRightsSnapshotMetadataModel.get_by_id(
+                snapshot_content_model.id))
+        snapshot_metadata_model.mentioned_user_ids = list(
+            set(reconstituted_rights_model.owner_ids) |
```

Comment:
Just a check -- do we need to audit / have we already audited to ensure that all these fields exist in all models / no value is None?